components/style/parallel.rs

```diff
@@ -68,21 +68,27 @@ fn top_down_dom<N, C>(unsafe_nodes: UnsafeNodeList,
         }
 
         // Perform the appropriate traversal.
-        context.process_preorder(node);
+        let should_stop = match context.process_preorder(node) {
+            RestyleResult::Stop => true,
+            RestyleResult::Continue => false,
+        };
```

Comment:
Why the temporary? Couldn't you do:

    if let RestyleResult::Continue = context.process_preorder(node) {
        ...
    }

Even if the temporary is necessary for borrow check reasons, I'd prefer this to be written as:

    let restyle_result = context.process_preorder(node);
    if let RestyleResult::Continue = restyle_result {
        ...
    }